src/main/java/seedu/bigpp/ui/UI.java

```diff
@@ -34,7 +34,11 @@ public static void updateUI() {
 
         switch (uiState) {
         case MAIN_MENU:
-            printMainMenu();
+            if (isStartingMainMenu()) {
+                printStartingMainMenu();
+            } else {
+                printMainMenu();
+            }
```

Comment:
this allows for user to use back feature from any where in the stack except the starting main menu